include/libkrun.h

```diff
@@ -113,6 +120,41 @@ int32_t krun_set_data_disk(uint32_t ctx_id, const char *disk_path);
  */
 int32_t krun_add_disk(uint32_t ctx_id, const char *block_id, const char *disk_path, bool read_only);
 
+/* Supported disk image formats */
+#define KRUN_DISK_FORMAT_UNSPECIFIED 0
+#define KRUN_DISK_FORMAT_RAW 1
+#define KRUN_DISK_FORMAT_QCOW2 2
```

Comment:
It'd probably be safer to make this an `enum`.

Reply:
Code style wise sure, this could be nicer. But how it is safer?

If we also change the `krun_add_disk2` argument to be the enum type this could get more unsafe on the ABI level. Because theoretically, you cannot control what backing type the enum is in C, which complicates the ffi on the rust side (or other language for that matter). I am not sure how likely this is to be a weird type and using an `i32` on the Rust side is very likely safe, but not guaranteed.

Relevant quotes in C spec: https://stackoverflow.com/a/1113869